Source/Rhetos.Deployment/Rhetos.Deployment.csproj

```diff
@@ -95,6 +97,10 @@
     <Compile Include="Sql.cs" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\..\CommonConcepts\Plugins\Rhetos.Dom.DefaultConcepts\Rhetos.Dom.DefaultConcepts.csproj">

```

Comment:
Rhetos.Deployment is part of the core framework, is should not directly reference a plugin CommonConcepts. The motivation behind this constraint is to make all plugins equal and to force us to make the core framework as flexible as possible.

Unfortunately it seems that the old way of processing arguments in Rhetos needs to be redesigned to allow a clean way to provide arguments to plugins.

For now, the simplest solution I can think of is to add this new parameter to the Argument class (as you already did), and use dependency injection to provide this class to KeepSynchronizedRecomputeOnDeploy: It is probably best to directly register the `Arguments` class in `DeployPackages.AutofacModuleConfiguration` and use it is as a constructor parameter in `KeepSynchronizedRecomputeOnDeploy`.